string name
Position pos
string last_action
string last_action_result
string[] last_action_params